### PR TITLE
WIP: add example of i18n for graphql DTO

### PR DESCRIPTION
**Pull request by HegelPro**

### CODE REVIEW

File: packages/codegen/src/building-blocks/util/parseTs.ts [hunk 1]
```diff
@@ -0,0 +1,8 @@
+import j from "jscodeshift";
+
+export function parseTs(source: string, options?: j.Options): j.Collection<any> {
+  return j(source, {
+    parser: require("recast/parsers/babel-ts"),
```
Comment (vadimbasko) [inline packages/codegen/src/building-blocks/util/parseTs.ts:5]: @HegelPro need comment, why we should use recast here. Links to doc also will be nice to have.

File: packages/codegen/src/building-blocks/stages/template-model/pieces/getEntityMessages.ts [hunk 2]
```diff
@@ -0,0 +1,26 @@
+import pluralize from "pluralize";
+import { GqlAttributeModel } from "./entity";
+
+export type EntityMessages<K extends string = string> = {
+  [P in K]: EntityMessages | string;
+};
+
+export function getEntityMessages(
```
Comment (vadimbasko) [inline packages/codegen/src/building-blocks/stages/template-model/pieces/getEntityMessages.ts:8]: Please add comment with function description. And unit test `getEntityMessages.test.ts`

File: packages/codegen/src/generators/react-admin/common/mergeI18nMessagesWithLocaleFile.ts [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+ */
+export async function mergeI18nMessagesWithLocaleFile(
+  screenPaths: ScreenPaths,
+  localeName: string,
+  defaultMessages: EntityMessages
+) {
+  const localePath = path.join(
+    screenPaths.localeMessagesPath,
+    `${localeName}.ts`
+  );
```
Comment (vadimbasko) [inline packages/codegen/src/generators/react-admin/common/mergeI18nMessagesWithLocaleFile.ts:27]: Not sure that we need so many lines for this code, read better in one line
```suggestion
  const localePath = path.join(screenPaths.localeMessagesPath, `${localeName}.ts`);
```

There are also several similar cases in this file.

File: packages/codegen/src/generators/react-admin/common/mergeI18nMessagesWithLocaleFile.ts [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+  localeName: string,
+  defaultMessages: EntityMessages
+): string {
+  const source = parseTs(localeFileContent);
+
+  const localeVariable = source.findVariableDeclarators(localeName);
+  const hasResourcesProperty =
+    localeVariable.find(j.ObjectProperty, {
+      key: { name: "resources" }
+    }).length === 0;
```
Comment (vadimbasko) [inline packages/codegen/src/generators/react-admin/common/mergeI18nMessagesWithLocaleFile.ts:60]:
```suggestion
  const hasResourcesProperty =
    localeVariable.find(j.ObjectProperty, {key: { name: "resources" }}).length === 0;
```

File: packages/codegen/src/generators/react-admin/common/mergeI18nMessagesWithLocaleFile.test.ts [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+      }
+    }`;
+    const localeResult = mergeI18nMessagesWithLocaleFileContent(locale, "en", {
+      AnyDTO: {
+        newField: "newValue",
+        oldTouchedField: "newValue"
+      }
+    });
+    expect(localeResult).to.equal(localeExpected);
+  });
```
Comment (vadimbasko) [inline packages/codegen/src/generators/react-admin/common/mergeI18nMessagesWithLocaleFile.test.ts:95]: If there is no 'en' object in file, error should be thrown
```suggestion
  it("should throw obvious error message if there is no 'en' object in file", function() {
    const locale = `
    export const incorrectLocalObjectName: TranslationMessages = {
      ...englishMessages,
    }`;
    // TODO Error should be thrown, error message should be clear
    mergeI18nMessagesWithLocaleFileContent(locale, "en", {});
  });
```

File: packages/codegen/src/generators/react-admin/common/mergeI18nMessagesWithLocaleFile.ts [hunk 6]
```diff
@@ -0,0 +1,124 @@
...
+  screenPaths: ScreenPaths,
+  localeName: string,
+  defaultMessages: EntityMessages
+) {
+  const localePath = path.join(
+    screenPaths.localeMessagesPath,
+    `${localeName}.ts`
+  );
+
+  const localeFile = await fs.readFile(localePath, "utf-8");
```
Comment (vadimbasko) [inline packages/codegen/src/generators/react-admin/common/mergeI18nMessagesWithLocaleFile.ts:29]: Add check, that 'localeFile' exist. If no - throw error with comprehensive message.

File: packages/codegen/src/generators/react-admin/common/mergeI18nMessagesWithLocaleFile.ts [hunk 7]
```diff
@@ -0,0 +1,124 @@
...
+import j from "jscodeshift";
+import * as path from "path";
+import { EntityMessages } from "../../../building-blocks/stages/template-model/pieces/getEntityMessages";
+import { parseTs } from "../../../building-blocks/util/parseTs";
+import { ScreenPaths } from "./generatorPaths";
+
+/**
+ * Find locale file by localeName in file system of generated app
+ * then find variable with localeMessages by localeName
+ * then find "resources" property inside variable
```
Comment (vadimbasko) [inline packages/codegen/src/generators/react-admin/common/mergeI18nMessagesWithLocaleFile.ts:12]:
```suggestion
 * then find "resources" property inside variable, if there is no 'resources' property - create it
```

File: packages/codegen/src/generators/react-admin/common/mergeI18nMessagesWithLocaleFile.ts [hunk 8]
```diff
@@ -0,0 +1,124 @@
...
+ * then merge default messages with existing messages to locale file in file system
+ * 
+ * @param screenPaths - contain information about i18n messages dir path
+ * @param localeName - using to get locale file in i18n messages dir and find variable with locale messages inside
+ * @param defaultMessages - messages that should by in generated app, them can be overridden later in generated app
+ */
+export async function mergeI18nMessagesWithLocaleFile(
+  screenPaths: ScreenPaths,
```
Comment (vadimbasko) [inline packages/codegen/src/generators/react-admin/common/mergeI18nMessagesWithLocaleFile.ts:20]: we don't need all `screenPaths` here, `localeMessagesPath` is enough

**Review by vadimbasko — CHANGES_REQUESTED** [review]
A few things should be clarified. Do not merge PR after fix, I will look on it once more after changes will be made.

File: packages/codegen/src/generators/react-admin/common/mergeI18nMessagesWithLocaleFile.ts (hunk 4 above)
Comment (HegelPro) [reply packages/codegen/src/generators/react-admin/common/mergeI18nMessagesWithLocaleFile.ts:60]: It's prettier formatter

File: packages/codegen/src/generators/react-admin/common/mergeI18nMessagesWithLocaleFile.ts (hunk 3 above)
Comment (HegelPro) [reply packages/codegen/src/generators/react-admin/common/mergeI18nMessagesWithLocaleFile.ts:27]: It's prettier formatter

Comment (vadimbasko) [reply packages/codegen/src/generators/react-admin/common/mergeI18nMessagesWithLocaleFile.ts:27]: Prettier is not standard for code in 'codegen' package. The only place where it must - is generated code. 
Code readability has higher priority than prettier formatting for code in 'codegen'. If you are using prettier formatting, please review your code after it. 
Also I can suggest increase line length in prettier settings (printWidth: 100). This value we are using in generated code formatting  <https://github.com/Amplicode/amplicode-frontend/blob/master/packages/codegen/src/building-blocks/stages/configuring/defaultConfigureGenerator.ts#L62>

**Review by vadimbasko — APPROVED** [review]
A few TODO added, but we will discuss them after merge.

**Pull request merged**